### PR TITLE
support an isolated bioconda-utils conda installation

### PR DESCRIPTION
**Pull request by daler**
* [ ] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [ ] This PR updates an existing recipe.
* [x] This PR does something else (explain below).

## Summary
`./simulate-travis.py --bootstrap /tmp/miniconda` makes it much more pleasant to work on bioconda recipes, and `simulate-travis.py` now more closely mimics exactly what is run on travis-ci.

## Rationale
Previously, contributors were forced to change their root conda environment to run tests locally. This is because `bioconda-utils` requires `conda-build`, which in turn will only work in the root environment. This was frustrating if the root environment was used for something other than bioconda development, and was brought up in #5134, https://github.com/bioconda/bioconda-utils/pull/145#issuecomment-312010030, and #5355.

## This PR
This PR adds the ability to install a separate version of conda, isolated from whatever may already be on the machine, and keeps track of that location during subsequent calls of `simulate-travis.py` -- all while being as minimally invasive as possible. I've changed the travis-ci setup to use this command as well, so local testing will now be a better simulation of the travis environment.

To start using it, first perform the setup:

```bash
./simulate-travis.py --bootstrap /tmp/miniconda 
```

That's it -- then use `simulate-travis.py` as before. You can even remove bioconda-utils and conda-build from your root environment. I've added some additional features as well, see below. 

## Other changes
- Simplified the steps to get started on recipes. All you need to start developing bioconda recipes is Python (2 or 3) and the PyYAML package. Optionally, docker. Everything else is handled automatically. This also simplifies setup on travis-ci.
- `~/.config/bioconda/config.yml` persistently stores the location of the bootstrapped installation so you don't have to specify each time
- Previously, mulled-build tests did not run locally when running `simulate-travis.py`. mulled-build will now run by default, but there is an argument to disable it
- `--git-range` argument to `simulate-travis.py` specifies a git range to be used for both linting and building
- `--skip-linting` argument for . . . skipping linting
- Documentation in bioconda-utils has been updated to reflect these changes.

### CODE REVIEW

**Comment by daler** [comment]
@bioconda/core can I get a review on this please? That last commit ead08ce builds a recipe using the new framework to test it, and that works. But it would be helpful just to get another set of eyes on this. Thanks.

File: simulate-travis.py [hunk 1]
```diff
@@ -101,30 +250,37 @@ def _remote_or_local(fn, branch='master', remote=False):
...
 
-    sp.run(
-        [
-            'pip', 'install',
-            'git+https://github.com/bioconda/bioconda-utils.git@{0}'.format(env['BIOCONDA_UTILS_TAG'])
-        ],
-        check=True)
+if args.install_alternative_conda:
+    _install_alternative_conda(args.install_alternative_conda, overwrite=args.overwrite)
+    sys.exit(0)
```
Comment (bgruening) [inline simulate-travis.py]: We could output here a small message, like conda environment successfully created at ...

File: simulate-travis.py [hunk 2]
```diff
@@ -101,30 +250,37 @@ def _remote_or_local(fn, branch='master', remote=False):
...
+
+if args.set_channel_order:
+    _set_channel_order()
     sys.exit(0)
 
+if args.bootstrap:
+    _install_alternative_conda(args.bootstrap, overwrite=args.overwrite)
+    _set_channel_order()
+    _install_requirements()
+    sys.exit(0)
```
Comment (bgruening) [inline simulate-travis.py]: Maybe here as well, to tell the user what was done.

File: simulate-travis.py [hunk 3]
```diff
@@ -156,10 +312,18 @@ def _remote_or_local(fn, branch='master', remote=False):
         ('--docker' not in env['BIOCONDA_UTILS_BUILD_ARGS'])
     ):
         env['DOCKER_ARG'] = '--docker'
+        if not args.disable_mulled:
+            env['DOCKER_ARG'] += ' --mulled-test'
 
     # Override env with whatever's in the shell environment
     env.update(os.environ)
+
+    # Only at the very end do we want to modify the path:
```
Comment (bgruening) [inline simulate-travis.py]: s/do//

**Comment by daler** [comment]
Thanks for reviewing Bjoern!

**Pull request merged**